app/src/main/java/com/scottyab/challenge/presentation/details/DetailsScreen.kt

```diff
@@ -0,0 +1,159 @@
+package com.scottyab.challenge.presentation.details
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.WindowInsets
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.text.BasicText
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Share
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SnackbarDuration
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.SnackbarResult
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import coil.compose.SubcomposeAsyncImage
+import com.scottyab.challenge.R
+import com.scottyab.challenge.presentation.details.DetailsUiEvent.ShareClick
+import com.scottyab.challenge.presentation.details.DetailsUiEvent.SnackMessageShown
+import com.scottyab.challenge.presentation.details.DetailsUiEvent.UpClick
+import com.scottyab.challenge.presentation.snapshots.model.SnapshotUi
+import com.scottyab.challenge.presentation.ui.SampleAppTheme
+import com.scottyab.challenge.presentation.ui.common.AppProgressIndicator
+import com.scottyab.challenge.presentation.ui.common.DefaultTopBar
+import java.time.LocalDateTime
+
+@Suppress("ktlint:standard:function-naming")
+@Composable
+fun DetailsScreen(
```

Comment:
New Composable for the Details screen. Built without dependancy on the Viewmodel to aid ComposeUI testing